### ISSUE TITLE
Make sure -X DELETE for contexts is synchronous

### ISSUE THREAD

**Issue by CBribiescas** [spark-jobserver/spark-jobserver#395]
According to the documentation, sync option is only for jobs

> Asynchronous and synchronous job API. Synchronous API is great for low latency jobs!

It would be nice to have a sync option for context creation.  Quickly (or programmatically) running the following repeatedly results in out of sync issues.

``````
curl -d "" -X POST -H "application/x-www-form-urlencoded" "http://localhost:8090/contexts/context?num-cpu-cores=4&memory-per-node=512m&sync=true"
curl -d "" -X DELETE -H "application/x-www-form-urlencoded" "http://localhost:8090/contexts/context?sync=true"````
``````

**Comment by velvia** [comment]
Hi @CBribiescas 

Context creation is always synchronous -- when the route comes back, that means the context is created.  DELETE might not be synchronous though - which we can look into.

**Comment by CBribiescas** [comment]
Yes, I tested them separately and it does look like only the delete is not synchronous.  Create operated correctly.

**Comment by noorul** [comment]
@CBribiescas, @velvia  What exactly is the issue here. When I create and delete context, I don't see any timing issue.

**Comment by bsikander** [comment]
I am also facing this issue. Since, DELETE is async, when you try to create a new context right after deleting it, you get a 400 response back. The temporary solution which I found in the source code of jobserver is to do a Thread.sleep. See line [131 of this file](https://github.com/spark-jobserver/spark-jobserver/blob/0919588387d85460460401556614230148ceeae5/job-server/src/test/scala/spark/jobserver/LocalContextSupervisorSpec.scala).

**Comment by emlagowski** [comment]
I also have to kill context, create new one and then run a job.

My queries and responses:
```
DELETE ctxName -> OK
POST create ctxName -> 'context ctxName exists'
POST run job on ctxName -> 'context ctxName not found'
```

The possibility of synchronous DELETE could be useful.

**Comment by doernemt** [comment]
@velvia issue can be closed, my PR to fix it was merged a month ago and is part of the 0.7 release. 
Thanks, Tim

**Comment by velvia** [comment]
@doernemt thanks
